src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -124,7 +137,19 @@ def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None
         if not ret:
             logger.info(f'{daemon.hostname} removed from nvmeof gateways dashboard config')
 
-        # and any certificates being used for mTLS
+        # Assert configured
+        assert self.pool
+        assert self.group is not None
+        # Notify monitor about this gateway deletion
+        cmd = {
+            'prefix': 'nvme-gw delete',
+            'id': daemon.hostname,
+            'group': self.group,
+            'pool': self.pool
+        }
+        _, _, err = self.mgr.mon_command(cmd)
+        if err:
+            self.mgr.log.error(f"Unable to send monitor command {cmd}, error {err}")
```

Comment:
This one is still just a log line. I don't know if the behavior should or should not be symmetric with the 'gateway create' operation in the other function. If it should be you probably should change this to raise an exception too.

Reply:
This does not appear to be a fatal error, I hope the log line about the error is enough here.

Reply:
:shrug: I don't know enough about the needs and design of nvmeof HA to agree or disagree - All I can say is if this command fails and something breaks because of it I'd not enjoy being the admin to have to scrounge through the logs to find this.